tests/integration/deploy_jenkins.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Deploy jenkins to lxd using juju."""
+
+import dataclasses
+import logging
+import subprocess
+import sys
+import time
+
+import yaml
+from selenium.webdriver import Firefox, FirefoxOptions
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+
+logging.basicConfig(
+    stream=sys.stdout, format="%(levelname)s %(asctime)s - %(message)s", level=logging.INFO
+)
+LOGGER = logging.getLogger()
+
+
+@dataclasses.dataclass
+class JenkinsDeployment:
+    """Key information about a jenkins deployment."""
+
+    controller_name: str
+    model_name: str
+    unit_number: int
+    public_address: str
+    username: str
+    password: str
+
+    @property
+    def hostname(self) -> str:
+        """Calculate the hostname."""
+        if ":" in self.public_address:
+            return f"[{self.public_address}]"
+        return self.public_address
+
+
+def jenkins_active() -> bool:
+    """Check whether the jenkins application is active."""
+    result = subprocess.check_output(["juju", "status"])
+    LOGGER.info("juju status \n%s", result.decode("utf-8"))
+    result = subprocess.check_output(["juju", "status", "--format", "yaml"])
+    return (
+        yaml.safe_load(result)["applications"]["jenkins"]["application-status"]["current"]
+        == "active"
+    )
+
+
+def deploy_jenkins() -> JenkinsDeployment:
+    """Deploys jenkins to the lxd controller."""
+    # Deploy
+    LOGGER.info("deploying jenkins")
+    result = subprocess.check_output(["juju", "controllers", "--format", "yaml"])
+    controller_name = next(
+        filter(
+            lambda item: item[1]["cloud"] == "localhost",
+            yaml.safe_load(result)["controllers"].items(),
+        )
+    )[0]
+    model_name = "jenkins"
+    subprocess.check_output(["juju", "switch", controller_name])
+    subprocess.check_output(["juju", "add-model", model_name, "localhost"])
+    subprocess.check_output(["juju", "deploy", "jenkins"])
```

Comment:
Once the model is switch ed, aren't you able to deploy with `ops_test.model.deploy`?

Reply:
This is the same as above, `setup_jenkins.py` is currently not run during integration tests, it runs to setup CI. Marking as resolved for now, feel free to reopen if you disagree.